core/src/main/java/io/opencensus/tags/TagValueString.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2017, Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.tags;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.base.Preconditions;
+import io.opencensus.internal.StringUtil;
+import io.opencensus.tags.TagKey.TagKeyString;
+import javax.annotation.concurrent.Immutable;
+
+/** A validated tag value associated with a {@link TagKeyString}. */
+@Immutable
+@AutoValue
+public abstract class TagValueString {
```

Comment:
For consistency should we do the same for TagValue -> TagValueString as we do for Key and Tag? also match will become:

public <T extends TagValue> T match(
    Function<? super TagString, TagStringValue> stringFunction,
    Function<? super TagLong, TagLongValue> longFunction,
    Function<? super TagBoolean, TagBooleanValue> booleanFunction);